tests/base/src/test/java/org/keycloak/test/admin/SMTPConnectionTest.java

```diff
@@ -15,53 +15,93 @@
 import org.keycloak.admin.client.resource.RealmResource;
+import org.keycloak.models.AdminRoles;
+import org.keycloak.models.Constants;
 import org.keycloak.representations.idm.RealmRepresentation;
-import org.keycloak.representations.idm.UserRepresentation;
-import org.keycloak.testsuite.AbstractKeycloakTest;
-import org.keycloak.testsuite.util.GreenMailRule;
-import org.keycloak.testsuite.util.UserBuilder;
 
 import jakarta.mail.internet.MimeMessage;
 import jakarta.ws.rs.core.Response;
+import org.keycloak.test.framework.annotations.InjectKeycloakUrls;
+import org.keycloak.test.framework.annotations.InjectRealm;
+import org.keycloak.test.framework.annotations.KeycloakIntegrationTest;
+import org.keycloak.test.framework.mail.MailServer;
+import org.keycloak.test.framework.mail.annotations.InjectMailServer;
+import org.keycloak.test.framework.realm.ManagedRealm;
+import org.keycloak.test.framework.realm.RealmConfig;
+import org.keycloak.test.framework.realm.RealmConfigBuilder;
+import org.keycloak.test.framework.server.KeycloakUrls;
+
 import java.util.HashMap;
-import java.util.List;
 import java.util.Map;
 
-import static org.junit.Assert.assertEquals;
-import static org.junit.Assert.fail;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.fail;
 import static org.keycloak.representations.idm.ComponentRepresentation.SECRET_VALUE;
 
 /**
  * @author <a href="mailto:bruno@abstractj.org">Bruno Oliveira</a>
  */
-public class SMTPConnectionTest extends AbstractKeycloakTest {
+@KeycloakIntegrationTest
+public class SMTPConnectionTest {
 
-    public final String SMTP_PASSWORD = setSmtpPassword();
+    @InjectRealm(config = SMTPRealmWithClientAndUser.class)
+    private ManagedRealm managedRealm;
 
-    @Rule
-    public GreenMailRule greenMailRule = new GreenMailRule();
     private RealmResource realm;
 
-    @Override
-    public void addTestRealms(List<RealmRepresentation> testRealms) {
+    @InjectMailServer
+    private MailServer mailServer;
+
+    @InjectKeycloakUrls
+    private KeycloakUrls keycloakUrls;
+
+    private Keycloak keycloak;
+
+    private static class SMTPRealmWithClientAndUser implements RealmConfig {
+
+        @Override
+        public RealmConfigBuilder configure(RealmConfigBuilder realm) {
+            realm.addClient("myclient")
+                    .secret("mysecret")
+                    .directAccessGrants();
+
+            realm.addUser("myadmin")
+                    .name("My", "Admin")
+                    .email("admin@localhost")
+                    .emailVerified()
+                    .password("myadmin")
+                    .clientRoles(Constants.REALM_MANAGEMENT_CLIENT_ID, AdminRoles.REALM_ADMIN);
+
+            return realm;
+        }
     }
 
-    public String setSmtpPassword() {
-        return "admin";
+    @BeforeEach
+    public void setupSMTP() {
+        keycloak = KeycloakBuilder.builder()
```

Comment:
This can be simplified a lot when we get this in https://github.com/keycloak/keycloak/pull/35693/files#diff-21086ccd1fef58bb212f988a6258dea56cd66f0735813643b1510a87315d92c1R33